spec/requests/demo_mode/sessions_spec.rb

```diff
@@ -77,32 +77,61 @@
       end
 
       context 'with option' do
-        before do
-          DemoMode.configure do
-            around_persona_generation do |generator, options|
-              generator.call.tap do |dummy_user|
-                if options.present? && options[:example_custom_option].present?
-                  dummy_user.update!(name: options[:example_custom_option][:name])
+        context 'and override with around_persona_generation' do
+          before do
+            DemoMode.configure do
+              around_persona_generation do |generator, options|
+                generator.call.tap do |dummy_user|
+                  if options.present? && options[:example_custom_option].present?
+                    dummy_user.update!(name: options[:example_custom_option][:name])
+                  end
                 end
               end
             end
           end
+
+          it 'creates a session and returns processing json saving the option on the created session' do
+            post '/ohno/sessions', params: {
+              session: { persona_name: 'the_everyperson' },
+              options: { example_custom_option: { name: 'Tester' } },
+            }.to_json, headers: request_headers
+
+            last_session = DemoMode::Session.last
+            perform_enqueued_jobs
+
+            expect(DummyUser.last.name).to eq 'Tester'
+            expect(response_json['id']).to eq last_session.id
+            expect(response_json['processing']).to be true
+            expect(response_json['username']).to be_nil
+            expect(response_json['password']).to be_nil
+          end
         end
 
-        it 'creates a session and returns processing json saving the option on the created session' do
-          post '/ohno/sessions', params: {
-            session: { persona_name: 'the_everyperson' },
-            options: { example_custom_option: { name: 'Tester' } },
-          }.to_json, headers: request_headers
+        context 'and override with sign_in_as persona' do
+          before do
+            DemoMode.add_persona :example_tester do
+              features << ""
+              sign_in_as do |**options|
+                DummyUser.create!(name: options.dig(:example_custom_option, :name) || 'Example Tester')
+              end
+            end
+          end
 
-          last_session = DemoMode::Session.last
-          perform_enqueued_jobs
+          it 'creates a session and returns processing json saving the option on the created session' do
+            post '/ohno/sessions', params: {
+              session: { persona_name: 'example_tester' },
+              options: { example_custom_option: { name: 'New Example Tester' } },
+            }.to_json, headers: request_headers
```

Comment:
@smudge is this what you were looking for?

Reply:
I am confused as well. I was looking for a way to pass the param through, and I see that in some places, we have an unused `_password` param. Can you point me to where the param should be passed into the correct way?

Reply:
So looks like clearing the `around_persona_generation` does reset the test and now it's failing correctly.
```
DemoMode.configure do
              around_persona_generation do |generator, options|
                generator.call(options)
              end
            end
```

Reply:
I have updated it to use a named argument instead.